components/ui-shell/index.stories.ts

```diff
@@ -13,7 +13,12 @@ export default {
   title: 'UI Shell',
 };
 
-export const Default = () => html`<qiskit-ui-shell></qiskit-ui-shell>`;
+export const Default = () =>
+  html`<qiskit-ui-shell
+    @on-click="${(e: { detail: { label: string; url: string } }) => {
```

Comment:
Could also use CustomEvent type to satisfy typescript

```suggestion
    @on-click="${(e: CustomEvent) => {
```